app/graphql/types/interfaces/rules_preload.rb

```diff
@@ -12,8 +12,8 @@ def rules(args = {})
     latest_test_result_batch(args).then do |latest_test_result|
       latest_rule_results_batch(latest_test_result).then do |rule_results|
         rules_for_rule_results_batch(rule_results).then do |rules|
-          initialize_rules_context(rules, rule_results, args)
-          rules
+          initialize_rules_context(rules.compact, rule_results, args)
```

Comment:
This is the culprit of the alerts we get - the method called does: `      context[:parent_profile_id][rule.id] = object.id` for a nil `rule`